torch/csrc/jit/tensorexpr/ir_printer.cpp

```diff
@@ -153,6 +153,12 @@ void IRPrinter::visit(const CompareSelect* v) {
   if (rhs_prec >= self_prec) {
     os() << ")";
   }
+  os() << " ? ";
+  os() << "(";
+  v->ret_val1()->accept(this);
```

Comment:
I think you just want to check if the precedence of `ret_val1` is lower than it is for CompareSelect to avoid the parens.

Reply:
and done!